overlord/snapstate/snapmgr.go

```diff
@@ -660,9 +678,11 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) error {
 	}
 
 	cand := snapst.Candidate
+	oldCurrent := snapst.Current
 
 	m.backend.Candidate(snapst.Candidate)
 	snapst.Sequence = append(snapst.Sequence, snapst.Candidate)
+	snapst.Current = snapst.Candidate.Revision
```

Comment:
nitpick, but I would put 

oldCurrent := snapst.Current
  
just before this line